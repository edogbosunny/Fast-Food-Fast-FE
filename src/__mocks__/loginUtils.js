/* eslint-disable import/prefer-default-export */
export const props = {
  auth: {
    isAuthenticated: true,
    loading: null,
    user: { id: 1, iat: 1549439277, exp: 1549525677 },
  },
  errors: {
    isAuthenticated: false,
    loading: null,
    email: '',
    password: '',
    message: '',
  },
  user: {},
  history: {
    action: 'PUSH',
    push: jest.fn(),
  },
  location: {
    hash: '',
    pathname: '/login',
    search: '',
  },
  loginUser: jest.fn(),
  match: { path: '/login', url: '/login', isExact: true },
};

export const mockedLoginState = {
  auth: {
    isAuthenticated: true,
    loading: null,
    user: { id: 1, iat: 1549439277, exp: 1549525677 },
  },
  errors: {
    isAuthenticated: false,
    loading: null,
    email: '',
    password: '',
    message: '',
  },
};

export const nextProps = {
  auth: {
    isAuthenticated: true,
    loading: null,
    user: { id: 1, iat: 1549439277, exp: 1549525677 },
  },
  history: {
    action: 'PUSH',
    push: jest.fn(),
  },
};

export const Loginstate = {
  email: '',
  password: '',
};
