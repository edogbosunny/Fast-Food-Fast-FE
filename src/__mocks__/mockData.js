const mock = {
  loginDetails: {
    email: 'test@test.com',
    password: '123456',
  },
  registerDetails: {
    email: 'test1@test.com',
    password: 123456,
    confirmpassword: 123456,
    firstname: 'test',
    lastname: 'test,',
  },
  errRegisterDetails: {
    email: 'test1@test.com',
    password: 123456,
    confirmpassword: 123456,
    firstname: 'test',

  },
};
export default mock;
