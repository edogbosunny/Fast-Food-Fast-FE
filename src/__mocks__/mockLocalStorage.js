let localStorage = {};

export default {
  setItem(key, value) {
    return Object.assign(localStorage, { [key]: value });
  },
  removeItem(key) {
    return delete localStorage[key];
  },
  getItem(key) {
    return localStorage[key];
  },
  clear() {
    localStorage = {};
  },
};
