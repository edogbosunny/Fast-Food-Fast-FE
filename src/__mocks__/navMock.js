/* eslint-disable import/prefer-default-export */
export const props = {
  auth: {
    isAuthenticated: true,
    loading: null,
    user: { id: 1, iat: 1549439277, exp: 1549525677 },
  },
  errors: {
    isAuthenticated: false,
    loading: null,
    email: '',
    password: '',
    message: '',
  },
  history: {
    action: 'PUSH',
    push: jest.fn(),
  },
  location: {
    hash: '',
    pathname: '/login',
    search: '',
  },
  logoutUser: jest.fn(),
  match: { path: '/login', url: '/login', isExact: true },
};

export const mockState = {
  auth: {
    isAuthenticated: true,
    loading: null,
    user: { id: 1, iat: 1549439277, exp: 1549525677 },
  },
};
