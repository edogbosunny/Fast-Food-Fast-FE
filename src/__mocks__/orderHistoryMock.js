/* eslint-disable import/prefer-default-export */
export const props = {
  dispatch: jest.fn(),
  orderHistory: {
    loading: false,
    orderHistory: {
      count: 30,
      mealItem: [{}],
      message: 'Food order retrieved succesfully',
    },
    getAllOrders: jest.fn(),
    getUserOrderById: jest.fn(),
  },
};

export const tableProps = {
  getAllOrders: jest.fn(),
  getUserOrderById: jest.fn(),
  orderHistory: {
    loading: false,
    orderHistory: {
      count: 30,
      data: [{}],
      message: 'Food order retrieved succesfully',
    },
  },
};
