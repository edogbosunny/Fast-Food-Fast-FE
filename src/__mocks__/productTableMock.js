/* eslint-disable import/prefer-default-export */
export const props = {
  cart: {
    count: 800,
    cart: {
      0: {
        meal_id: 1, price: '600', meal: 'Spaniish Noodles', qty: 1,
      },
    },
    items: null,
    cartCount: 800,
  },
  checkoutOrder: jest.fn(),
  order: {
    order: {},
  },
  history: {
    push: jest.fn(),
  },
  removeFromCart: jest.fn(),
};
