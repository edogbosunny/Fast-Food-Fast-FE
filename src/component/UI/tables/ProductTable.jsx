import React from 'react';
import '../../../styles/ind.css';

const ProductTable = () => (
  <div className='yoo'>
    <table>
      <thead className='cart_summary'>
        <tr>
          <th>Your Order</th>
          <th />

        </tr>
      </thead>
      <tbody id='cartTable' />
      <thead className='cart_summary'>
        <tr>
          <th>
                        Total
          </th>
          <th id='grandTotal'>
                        0
          </th>
        </tr>
        <tr>
          <th>
            <div className='btnCenter'>
              <button type='button' className='checkoutBut'>Proceed to Checkout</button>
            </div>
          </th>
          <th />
        </tr>
      </thead>
    </table>
  </div>
);

export default ProductTable;
