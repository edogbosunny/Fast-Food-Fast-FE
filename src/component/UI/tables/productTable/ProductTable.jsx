/* eslint-disable react/no-unused-state */
/* eslint-disable import/no-named-as-default */
/* eslint-disable no-shadow */
/* eslint-disable react/prefer-stateless-function */
/* eslint-disable react/prop-types */
/* eslint-disable react/destructuring-assignment */
import React, { Component } from 'react';
import { connect } from 'react-redux';
import { withAlert } from 'react-alert';
import { withRouter } from 'react-router-dom';
import '../../../../styles/ind.css';
import TableComponent from './TableComponent';
import { checkoutOrder } from '../../../../actions/checkoutAction';

export class ProductTable extends Component {
  constructor() {
    super();
    this.onClick = this.onClick.bind(this);
  }

  onClick() {
    const mealOrder = {
      mealId: '',
      quantity: '',
    };

    const token = window.localStorage.getItem('jwtToken');
    const count = window.localStorage.getItem('count');

    if (count === '0' || !count) {
      this.props.alert.error('PLEASE ADD ITEM TO CART !');
    } else {
      this.props.alert.success('ORDER PLACED SUCCESSFULLY !');
    }
    this.props.history.push('/checkout');
    if (!token) {
      this.props.history.push('/login');
    }
    Object.values(this.props.cart.cart).map((orderData) => {
      mealOrder.mealId = (orderData.meal_id);
      mealOrder.quantity = JSON.stringify(orderData.meal_id);
      const { checkoutOrder } = this.props;
      return checkoutOrder(mealOrder);
    });
  }

  render() {
    // console.log('count---->', this.props);
    return (
      <div className='yoo'>
        <table className="table table-bordered">
          <thead className='thead-dark'>
            <tr>
              <th scope="col">Order</th>
              <th scope="col">Cost</th>
            </tr>
          </thead>
          <tbody>
            <TableComponent />
            <tr>
              <th>Total</th>
              <td data-label='Total'>{this.props.cart.count}</td>
            </tr>
            <tr colSpan='2'>
              <td>
                <button type='button' onClick={this.onClick} className='btn btn-success btn-block'>Place Order</button>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    );
  }
}


export const mapStateToProps = state => ({
  cartCount: state.cart.count,
  cart: state.cart,
  order: state.order,
});

export default connect(mapStateToProps, { checkoutOrder })(withRouter(withAlert(ProductTable)));
