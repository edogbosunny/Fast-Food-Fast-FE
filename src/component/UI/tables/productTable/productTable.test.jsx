import React from 'react';
import { shallow } from 'enzyme';
import { props } from '../../../../__mocks__/productTableMock';
import { ProductTable, mapStateToProps } from './ProductTable';

describe('orderHistore component test-suit', () => {
  const wrapper = shallow(<ProductTable {...props} />);
  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should test mapStateToProps', () => {
    const state = mapStateToProps(props);
    expect(state).toBeTruthy();
  });

  it('should test onClick', () => {
    // const spy = jest.spyOn(ProductTable.prototype, 'onClick');
    wrapper.find('button').at(0).simulate('click');
    const click = jest.fn(wrapper.instance().onSubmit);
    const promise = new Promise((resolve) => {
      resolve(wrapper.instance().onSubmit);
    });
    promise.then(() => expect(click).toHaveBeenCalledTimes(1));
  });
});
