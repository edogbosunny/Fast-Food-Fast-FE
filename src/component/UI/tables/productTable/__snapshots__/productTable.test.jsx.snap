// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`orderHistore component test-suit should match snapshot 1`] = `
<div
  className="yoo"
>
  <table
    className="table table-bordered"
  >
    <thead
      className="thead-dark"
    >
      <tr>
        <th
          scope="col"
        >
          Order
        </th>
        <th
          scope="col"
        >
          Cost
        </th>
      </tr>
    </thead>
    <tbody>
      <Connect(TableComponent) />
      <tr>
        <th>
          Total
        </th>
        <td
          data-label="Total"
        >
          800
        </td>
      </tr>
      <tr
        colSpan="2"
      >
        <td>
          <button
            className="btn btn-success btn-block"
            onClick={[Function]}
            type="button"
          >
            Place Order
          </button>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
